clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -282,6 +274,20 @@ private long appendEosRecord(long offset, long timestamp, ByteBuffer key, ByteBu
         return crc;
     }
 
+    private long appendOldRecord(long offset, long timestamp, ByteBuffer key, ByteBuffer value) throws IOException {
```

Comment:
The trouble is that it is not only V0, but also V1, so we have to write something like `appendRecordV0OrV1`, or something awkward like that. I agree we should be consistent with however we name the `LogEntry` classes though.